src/bin-utils/get-script-by-prefix.test.js

```diff
@@ -0,0 +1,224 @@
+          foo: {
+            description: 'Foo script',
+            script: 'echo foo',
+          },
+        },
+      },
+    }
+    const actual = getScriptByPrefix(config, 'w.f')
+    const expected = {
+      script: 'echo foo',
+      name: 'watch.foo',
+      description: 'Foo script',
+    }
+    expect(actual).toEqual(expected)
+  },
+)
+
+test('getScriptByPrefix resolves ambiguities with single prefix', () => {
+  const config = {
+    scripts: {
+      foo: {
+        js: {
+          default: 'foo.js script',
+          script: 'echo foo.js',
+        },
+      },
+      foobar: {
+        description: 'Foobar script',
+        script: 'echo foobar',
+      },
+    },
+  }
+  const actual = getScriptByPrefix(config, 'f')
+  const expected = {
+    script: 'echo foobar',
+    name: 'foobar',
+    description: 'Foobar script',
+  }
+  expect(actual).toEqual(expected)
+})
+
+describe('getScriptByPrefix resolves ambiguities with nested prefix', () => {
+  let config
+  beforeEach(() => {
+    config = {
+      scripts: {
+        foo: {
+          bar: {
+            default: 'echo foo.bar',
+            baz: {
+              default: 'echo foo.bar.baz',
+              joe: {
+                default: {
+                  script: 'echo foo.bar.baz.joe',
+                },
+              },
+              invalid: 42,
+              s: {
+                t: 'echo t',
+              },
+            },
+          },
+        },
+        foobar: {
+          description: 'Foobar script',
+          script: 'echo foobar',
+        },
+      },
+    }
+  })
+  test('Foobar works correctly', () => {
+    const actual = getScriptByPrefix(config, 'f')
+    const expected = {
+      script: 'echo foobar',
+      name: 'foobar',
+      description: 'Foobar script',
+    }
+    expect(actual).toEqual(expected)
+  })
+  test('foo.bar works correctly', () => {
+    const actual = getScriptByPrefix(config, 'f.b')
+    const expected = {
+      script: 'echo foo.bar',
+      name: 'foo.bar.default',
+      description: '',
+    }
+    expect(actual).toEqual(expected)
+  })
+  test('foo.bar.baz works correctly', () => {
+    const actual = getScriptByPrefix(config, 'f.b.b')
+    const expected = {
+      script: 'echo foo.bar.baz',
+      name: 'foo.bar.baz.default',
+      description: '',
+    }
+    expect(actual).toEqual(expected)
+  })
+  test('foo.bar.baz.joe works correctly', () => {
+    const actual = getScriptByPrefix(config, 'f.b.b.j')
+    const expected = {
+      script: 'echo foo.bar.baz.joe',
+      name: 'foo.bar.baz.joe.default',
+      description: '',
+    }
+    expect(actual).toEqual(expected)
+  })
+  test('foo.bar.baz.x returns null', () => {
+    const actual = getScriptByPrefix(config, 'f.b.b.x')
+    expect(isNull(actual)).toBeTruthy()
+  })
+  test('f.x returns null', () => {
+    const actual = getScriptByPrefix(config, 'f.x')
+    expect(isNull(actual)).toBeTruthy()
+  })
+  test('d as prefix matches default', () => {
```

Comment:
Actually, I realize that this is the current behavior, but could we break that? That's one of my least favorite parts of `nps`. Because if there was a script by the name of `foo.bar.baz.define` and I tried the prefix `foo.bar.baz.def` it would actually pick up the default, which is _not_ what I'd want to have happen (if I wanted to run the default then I'd just do `foo.bar.baz`). So if you could break that, I'd really appreciate it.

Thanks!